iwyu_tool.py

```diff
@@ -176,19 +178,26 @@ def main(compilation_db_path, source_files, verbose, formatter, jobs, iwyu_args)
 
     # Run analysis
     try:
+        pending = []
+        while entries or pending:
```

Comment:
I think this could do with a little more abstraction to separate the process scheduling from the IWYU details. I have a half-baked `ProcessPool` locally, I can try and clean that up and post it. Thanks for showing me a basic approach, though, that helped me get somewhere with what I had!

Reply:
Since the whole thing was 20 LOC, i figured i'd just put it inline.

Reply:
Yeah, I think that makes sense. My attempt at this turned into an almost complete rewrite of `iwyu_tool.py`, and that's a little too revolutionary for my taste. Let me get a few refactors in to make your patch easier.